# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1386716657 18000
#      Tue Dec 10 18:04:17 2013 -0500
# Branch stable
# Node ID 60dee9205bb577874b0aca64ca2a086da71d233b
# Parent  3e914fbd1a0ec846f87609e72958e786d5324447
set default warning state before modifying for --traditional option

* octave.cc (octave_initialize_interpreter): Call octave
initialize_default_warning_state before maximum_braindamage.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -715,16 +715,21 @@ octave_initialize_interpreter (int argc,
 
   octave_program_invocation_name = octave_env::get_program_invocation_name ();
   octave_program_name = octave_env::get_program_name ();
 
   octave_thread::init ();
 
   set_default_prompts ();
 
+  // Initialize default warning state before --traditional option may
+  // reset them.
+
+  initialize_default_warning_state ();
+
   if (traditional)
     maximum_braindamage ();
 
   init_signals ();
 
   octave_ieee_init ();
 
   // The idea here is to force xerbla to be referenced so that we will
@@ -732,18 +737,16 @@ octave_initialize_interpreter (int argc,
   // library.  But octave_NaN should never be -1, so we should never
   // actually call xerbla.
 
   if (octave_NaN == -1)
     F77_FUNC (xerbla, XERBLA) ("octave", 13 F77_CHAR_ARG_LEN (6));
 
   initialize_error_handlers ();
 
-  initialize_default_warning_state ();
-
   if (! embedded)
     install_signal_handlers ();
   else
     quit_allowed = false;
 
   initialize_file_io ();
 
   install_types ();
